test/Whitelist.test.ts

```diff
@@ -0,0 +1,50 @@
+import {MockERC20Instance, WhitelistInstance} from '../build/types/truffle-types'
+
+const {expectEvent, expectRevert} = require('@openzeppelin/test-helpers')
+
+const MockERC20 = artifacts.require('MockERC20.sol')
+const Whitelist = artifacts.require('Whitelist.sol')
+// address(0)
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000'
+
+contract('Whitelist', ([owner, random]) => {
+  // ERC20 mocks
+  let usdc: MockERC20Instance
+  // Whitelist module
+  let whitelist: WhitelistInstance
+
+  const underlyingAsset = ZERO_ADDR
+
+  before('Deployment', async () => {
+    // deploy USDC token
+    usdc = await MockERC20.new('USDC', 'USDC')
+
+    // deploy Whitelist module
+    whitelist = await Whitelist.new({from: owner})
+  })
+
+  describe('Whitelist product', () => {
+    it('should revert whitelisting a product from non-owner address', async () => {
+      await expectRevert(
+        whitelist.whitelistProduct(underlyingAsset, usdc.address, usdc.address, {from: random}),
+        'Ownable: caller is not the owner',
+      )
```

Comment:
can we test that ownership can be transferred and that old owner can't whitelist and new owner can after that point?

Reply:
Will add it, but IMO it is out of the scope of this contract, as we are using OZ `Ownable` and that is already tested there.

Reply:
I agree with Haythem, we don't need that.